src/par/FixColumnPartitions.c

```diff
@@ -69,12 +74,22 @@ int FixColumnPartitions_UpDownFaces(Mesh_ptr mesh, MRegion_ptr mr, MFace_ptr* up
   while (found < 2) {
     rf = List_Entry(rfaces,i);
     if (!FixColumnPartitions_IsSideFace(mesh, mr, rf)) {
-      if (found < 1) *up = rf;
-      else *dn = rf;
+      if (up_unknown == 0) {
```

Comment:
Looks like you are accounting for the case when the top or bottom cell is degenerate. It would be good to have a comment somewhere to explain that